Microsoft.Toolkit.Mvvm/ComponentModel/Attributes/ObservablePropertyAttribute.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma warning disable CS1574
+
+using System;
+using System.ComponentModel;
+
+namespace Microsoft.Toolkit.Mvvm.ComponentModel
+{
+    /// <summary>
+    /// An attribute that indicates that a given field should be wrapped by a generated observable property.
+    /// In order to use this attribute, the containing type has to implement the <see cref="INotifyPropertyChanged"/> interface
+    /// and expose a method with the same signature as <see cref="ObservableObject.OnPropertyChanged(string?)"/>. If the containing
+    /// type also implements the <see cref="INotifyPropertyChanging"/> interface and exposes a method with the same signature as
+    /// <see cref="ObservableObject.OnPropertyChanging(string?)"/>, then this method will be invoked as well by the property setter.
+    /// <para>
+    /// This attribute can be used as follows:
+    /// <code>
+    /// partial class MyViewModel : ObservableObject
+    /// {
+    ///     [ObservableProperty]
+    ///     private string name;
```

Comment:
What happens if I want something camel cased like `IsEnabled`, if my field is named `isenabled` won't it generate `Isenabled`? Should we have another example for this if I need to define the field as `isEnabled` which seems a bit odd too.

Reply:
The property name generation logic is defined here:

https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/16ba7cc0c64310d0e82d04239183eece01161e7a/Microsoft.Toolkit.Mvvm.SourceGenerators/ComponentModel/ObservablePropertyGenerator.cs#L446-L460

That is correct, if your field is named `isenabled` then the resulting property will indeed be called `Isenabled`, but then again, one should never be calling anything `isenabled` anyway because that's against the naming guidelines 😄
This logic supports all common naming patterns, so `_fieldName`, `m_fieldName` and `fieldName`.

Reply:
[4th Bullet](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-properties) "Is", "Can", "Has" are common property name starters. So yes, it should be like `_isEnabled` I guess. I think we should at least have an example in the comment that shows the CamelCase scenario though, so that folks can get the properly cased Property name out of the generator.

Reply:
I would expect that people using `_snakeCase` would already have the right casing in place and it'd "just work", given that nobody in C# lang does ever really _not_ use upper casing to distinguish words in any given symbol name. That said sure, we could add some remarks in the XML docs to call out the currently supported formats, would that work? 😄

Reply:
Yeah, sounds great! Thanks! I think XML doc with the handled scenarios, plus another example property with the different syntax showing two words for CamelCasing the final property would cover all the bases.

Reply:
Done in https://github.com/CommunityToolkit/WindowsCommunityToolkit/pull/3873/commits/37caf88b68463ebb23b67c2e5d3f095485c9ecc5, also rebased to fix conflicts 😄